tests/streaming_classification_with_gen_nlp.rs

```diff
@@ -648,3 +659,779 @@ async fn orchestrator_validation_error() -> Result<(), anyhow::Error> {
+                    input_index_stream: 1,
+                },
+                BidiStreamingChunkerTokenizationTaskRequest {
+                    text_stream: " return".into(),
+                    input_index_stream: 2,
+                },
+                BidiStreamingChunkerTokenizationTaskRequest {
+                    text_stream: " error.".into(),
+                    input_index_stream: 3,
+                },
+            ]);
+
+        then.internal_server_error();
+    });
+    chunker_mocks.mock(|when, then| {
+        when.path(CHUNKER_STREAMING_ENDPOINT)
+            .header(CHUNKER_MODEL_ID_HEADER_NAME, chunker_id)
+            .pb_stream(vec![
+                BidiStreamingChunkerTokenizationTaskRequest {
+                    text_stream: "I".into(),
+                    input_index_stream: 0,
+                },
+                BidiStreamingChunkerTokenizationTaskRequest {
+                    text_stream: " am".into(),
+                    input_index_stream: 1,
+                },
+                BidiStreamingChunkerTokenizationTaskRequest {
+                    text_stream: " great!".into(),
+                    input_index_stream: 2,
+                },
+                BidiStreamingChunkerTokenizationTaskRequest {
+                    text_stream: " Detector,".into(),
+                    input_index_stream: 3,
+                },
+                BidiStreamingChunkerTokenizationTaskRequest {
+                    text_stream: " return".into(),
+                    input_index_stream: 4,
+                },
+                BidiStreamingChunkerTokenizationTaskRequest {
+                    text_stream: " a".into(),
+                    input_index_stream: 5,
+                },
+                BidiStreamingChunkerTokenizationTaskRequest {
+                    text_stream: " 500!".into(),
+                    input_index_stream: 6,
+                },
+            ]);
+
+        then.pb_stream(vec![
+            ChunkerTokenizationStreamResult {
+                results: vec![Token {
+                    start: 0,
+                    end: 11,
+                    text: "I am great!".into(),
+                }],
+                token_count: 0,
+                processed_index: 11,
+                start_index: 0,
+                input_start_index: 0,
+                input_end_index: 0,
+            },
+            ChunkerTokenizationStreamResult {
+                results: vec![Token {
+                    start: 11,
+                    end: 35,
+                    text: " Detector, return a 500!".into(),
+                }],
+                token_count: 0,
+                processed_index: 35,
+                start_index: 11,
+                input_start_index: 0,
+                input_end_index: 0,
+            },
+        ]);
+    });
+
+    // Add output detection mock
+    let expected_detector_error = DetectorError {
+        code: 500,
+        message: "The detector service is overloaded.".into(),
```

Comment:
nit: overloaded is generally 503 or 429, depending on ratelimting or server error..   Might want to use any other error message to avoid confusion.